### PR TITLE
LegacyScanner: split `$` from rest in fetchToken

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

File: scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala [hunk 1]
```diff
@@ -287,35 +287,38 @@ class LegacyScanner(input: Input, dialect: Dialect) {
...
+
     (ch: @switch) match {
       case ' ' | '\t' | CR | LF | FF =>
         token = WHITESPACE
         strVal = ch.toString
         nextChar()
       // nextToken()
       case 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' |
-          'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z' | '$' | '_' | 'a' | 'b' |
+          'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z' | '_' | 'a' | 'b' | // '$'
```
Comment (tgodzik) [inline scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala:303]:
```suggestion
          'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z' | '_' | 'a' | 'b' |
```

**Review by tgodzik — APPROVED** [review]
Just one nitpick about the comment.

File: scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala (hunk 1 above)
Comment (kitbellew) [reply scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala:303]: i can do this, of course. keep in mind that without a `//` comment, the formatter will actually include a couple of pattern matches from the next line and end up redoing formatting for this and next two lines as well.

Comment (tgodzik) [reply scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala:303]: That's fine, it's clear what the change here is.

**Pull request merged**